### PR TITLE
Add RequestOptions for low level clients

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+     * Adds a header to the HTTP request.
+     * @param header the header key
+     * @param value the header value
+     *
+     * @return the modified RequestOptions object
+     */
+    public RequestOptions addHeader(String header, String value) {
+        this.requestCallback = this.requestCallback.andThen(request ->
```
Comment (alzimmermsft) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java:94]: With this pattern if a new `requestCallback` is set won't all added headers be lost?

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    /**
+     * Adds a query parameter to the request URL.
+     * @param parameterName the name of the query parameter
+     * @param value the value of the query parameter
+     * @return the modified RequestOptions object
+     */
+    public RequestOptions addQueryParam(String parameterName, String value) {
+        this.requestCallback = this.requestCallback.andThen(request -> {
```
Comment (alzimmermsft) [inline sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java:106]: Same comment as `addHeader`

**Review by alzimmermsft — CHANGES_REQUESTED** [review]

File: sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java (hunk 1 above)
Comment (jianghaolu) [reply sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/http/RequestOptions.java:94]: changed to `addRequestCallback`

**Comment by jianghaolu** [comment]
Close via #22334

**Pull request closed without merging**